aten/src/ATen/TensorMeta.h

```diff
@@ -26,6 +26,9 @@ namespace impl {
 #define TORCH_META_FUNC(name) void structured_##name::meta
 #define TORCH_META_FUNC2(name, overload) void structured_##name##_##overload::meta
 
+#define TORCH_PRECOMPUTE_META_FUNC(name) structured_##name::precompute_out structured_##name::meta
+#define TORCH_PRECOMPUTE_META_FUNC2(name, overload) structured_##name##_##overload::precompute_out structured_##name##_##overload::meta
```

Comment:
I wonder if there actually is any reason to change the naming convention here, since, IIUC, precompute meta and meta are mutually exclusive (so it wouldn't be like you would have a naming conflict)

Reply:
reupping

Reply:
Which naming convention did I change? New macros are required because the return type for meta functions of kernels that have precompute are different.

Reply:
oh true. I guess we could unconditionally define `structured_##name::precompute_out` (maybe as `return_type`) and then just typedef it to void when it's not precomputed he he

Reply:
After your last comment, I don't feel too strongly about this, and wouldn't block on this